lib/utils/useWeb3Modal/index.tsx

```diff
@@ -29,52 +28,112 @@ export const useWeb3Modal = (strings: Web3ModalStrings): Web3ModalState => {
     }
   };
 
-  const downloadModal = async () => {
-    const newModal = (await import("../getWeb3Modal")).getWeb3Modal(strings);
```

Comment:
This was from my recent PR #789 and it removes ~800kb of dependencies from the initial page load, so we need to keep this `download` behavior. Or do a quick follow up PR to enhance w/ lazyloading.

Now that we wrote the modal component ourselves instead of relying on web3modal, we can probably use [Next Dynamic Imports w/ suspense](https://nextjs.org/docs/advanced-features/dynamic-import#example)

requirements:
* [ ] All references to web3 provider libraries (coinbase, torus, walletconnect) are contained in the `<ConnectModal />` component.
* [ ] when the page mounts it should lazyload in the background, even if the modal is not triggered/visible
* [ ] by the time the user clicks "connect" to open the modal, it should be downloaded
* [ ] maybe support a loading placeholder modal via react suspense, e.g. on slow connections where they click Connect before it can finish downloading

Reply:
added dynamic imports of specific libraries in useWeb3Modal as discussed on call.